frontend/src/components/pages/RegistrantExperience/PersonalInfo/CamperCard.tsx

```diff
@@ -0,0 +1,327 @@
+import React, { useEffect, useState } from "react";
+import { DeleteIcon } from "@chakra-ui/icons";
+import {
+  AlertDialog,
+  AlertDialogBody,
+  AlertDialogContent,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogOverlay,
+  Box,
+  Button,
+  Divider,
+  Flex,
+  FormControl,
+  FormLabel,
+  Heading,
+  Input,
+  NumberInputField,
+  NumberInput,
+  Spacer,
+  Text,
+  Textarea,
+  useDisclosure,
+  useToast,
+  Wrap,
+  WrapItem,
+  FormErrorMessage,
+} from "@chakra-ui/react";
+import {
+  PersonalInfoActions,
+  PersonalInfoReducerDispatch,
+} from "../../../../types/PersonalInfoTypes";
+import RequiredAsterisk from "../../../common/RequiredAsterisk";
+import { RegistrantExperienceCamper } from "../../../../types/CamperTypes";
+import { checkAge, checkFirstName, checkLastName } from "./personalInfoReducer";
+
+type CamperCardProps = {
+  nextBtnRef: React.RefObject<HTMLButtonElement>;
+  camper: RegistrantExperienceCamper;
+  camperId: number;
+  setPersonalInfo: (action: PersonalInfoReducerDispatch) => void;
+};
+
+const CamperCard = ({
+  nextBtnRef,
+  camper,
+  camperId,
+  setPersonalInfo,
+}: CamperCardProps): React.ReactElement => {
+  const [isFirstNameInvalid, setIsFirstNameInvalid] = useState<boolean>(false);
+  const [isLastNameInvalid, setIsLastNameInvalid] = useState<boolean>(false);
+  const [isAgeInvalid, setIsAgeInvalid] = useState<boolean>(false);
+
+  useEffect(() => {
+    const updateFormErrorMsgs = () => {
+      console.log(
+        checkFirstName(camper.firstName),
+        checkLastName(camper.lastName),
+        checkAge(camper.age),
+        camper.firstName,
+        !!camper.firstName,
+      );
+      if (!checkFirstName(camper.firstName)) setIsFirstNameInvalid(true);
+      if (!checkLastName(camper.lastName)) setIsLastNameInvalid(true);
+      if (!checkAge(camper.age)) setIsAgeInvalid(true);
+    };
+
+    if (nextBtnRef && nextBtnRef.current) {
+      // Passing the same reference
+      nextBtnRef.current.addEventListener("click", updateFormErrorMsgs);
+    }
+
+    return () => {
+      // Passing the same reference
+      if (nextBtnRef && nextBtnRef.current) {
+        nextBtnRef.current.removeEventListener("click", updateFormErrorMsgs);
+      }
+    };
+  }, [camper]);
+
+  function AlertDialogExample() {
+    const toast = useToast();
+    const { isOpen, onOpen, onClose } = useDisclosure();
+    const cancelRef = React.useRef(null);
+
+    return (
+      <>
+        <Button
+          color="text.critical.100"
+          variant="outline"
+          onClick={onOpen}
+          colorScheme="blue"
+        >
+          <DeleteIcon />
+          <Text textStyle={{ sm: "xSmallBold", lg: "buttonSemiBold" }} pl={3}>
+            Remove
+          </Text>
+        </Button>
+
+        <AlertDialog
```

Comment:
Any reason not to use modal here instead of alert dialog? Could we also shift this component to a new file?

Reply:
Oh, the reason was that I didn't know modal even existed in Chakra UI. I noticed that Ryan Change created a custom `DeleteModel.ts` component. Let me use that.

Reply:
And yep, let me write this into it's own file.